src/Processors/Formats/Impl/PrettyBlockOutputFormat.cpp

```diff
@@ -397,14 +432,24 @@ static String highlightDigitGroups(String source)
 
 void PrettyBlockOutputFormat::writeValueWithPadding(
     const IColumn & column, const ISerialization & serialization, size_t row_num,
-    size_t value_width, size_t pad_to_width, size_t cut_to_width, bool align_right, bool is_number)
+    size_t value_width, size_t pad_to_width, size_t cut_to_width, bool align_right, bool is_number, bool & has_break_line)
 {
     String serialized_value = " ";
     {
         WriteBufferFromString out_serialize(serialized_value, AppendModeTag());
         serialization.serializeText(column, row_num, out_serialize, format_settings);
     }
 
+    size_t break_line_pos = String::npos;
+    if (cut_to_width)
+        break_line_pos = serialized_value.find_first_of('\n');
+    if (break_line_pos != String::npos)
+    {
+        has_break_line = true;
+        serialized_value = serialized_value.substr(0, break_line_pos);
+        value_width = serialized_value.size() - 1;
```

Comment:
This doesn't look correct for several reasons.

1. The other lines can be longer than the first.
2. `size` is not a correct way to calculate the width.

Reply:
Let's add tests with:

```
тест
world
```

```
world
тест
```

```
test
\ttest
```